### PR TITLE
Update ava

### PR DESCRIPTION
**Pull request by ratson**
After I upgraded `ava`, I find a regression, which is probably a bug.

### CODE REVIEW

File: lib/collection.js [hunk 1]
```diff
@@ -525,6 +525,10 @@ Collection.prototype.findAndModify = function (query, update, opts, fn) {
  */
 
 Collection.prototype.findOne = function (query, opts, fn) {
+  if (typeof query === 'undefined') {
```
Comment (mathieudutour) [inline lib/collection.js]: that's unrelated right?

**Comment by ratson** [comment]
@mathieudutour It fixes the regression for https://github.com/Automattic/monk/blob/master/test/collection.js#L159, previously the test was passed instead of failed because `ava` did not check un-reached  statement.

**Comment by mathieudutour** [comment]
no but why `db.findOne()` should fail? it's perfectly valid

**Comment by ratson** [comment]
I agree, however the test said "findOne > findOne(undefined) should not work", what is your suggestion? The test will fail with the latest `ava` without change.

**Comment by mathieudutour** [comment]
oh I believe it's a typo, `findOne(undefined)` should work

**Comment by ratson** [comment]
Really? The test body do assertion in `catch`, I thought it is feature, because it is hard to imagine what should be returned by `findOne(undefined)`.

**Comment by mathieudutour** [comment]
yes actually you are right, there is an issue stating that it shoudn't work: https://github.com/Automattic/monk/issues/34

I'm not sure why, I wasn't contributing to the repo at that time. I thing that it should work since it's working in the mongo shell

**Comment by ratson** [comment]
I am open for this, change `findOne(undefined)` to be same as `findOne({})`?

**Comment by mathieudutour** [comment]
yes, I think that's fine

**Comment by ratson** [comment]
PR updated

**Comment by mathieudutour** [comment]
awesome thanks

**Comment by ratson** [comment]
@mathieudutour I think the travis test is failing because the cache, would you try clear the cache and re-run the build see if it helps?

**Pull request merged**